modin/backends/pandas/query_compiler.py

```diff
@@ -1180,6 +1180,52 @@ def sort_index(self, **kwargs):
         )
         return self.__constructor__(new_modin_frame)
 
+    def melt(self, *args, **kwargs):
+        def _convert_to_list(x):
+            if is_list_like(x):
+                x = [*x]
+            elif x is not None:
+                x = [x]
+            else:
+                x = []
+            return x
+
+        row_lengths = self._modin_frame._row_lengths
```

Comment:
Let's try to figure out a way to avoid accessing and relying on this field

Reply:
reimplemented